src/locked.message.coffee

```diff
@@ -0,0 +1,35 @@
+_ = require "lodash"
+debug = require("debug") "azure-servicebus-reader:locked-message"
+highland = require "highland"
+
+module.exports =
+  class TouchableMessage
+
+    constructor: (@service, message, opts = {}) ->
+      @lockedMessage = message
+      _.assign @, message
+      @timeToPeek = opts.timeToPeek or process.env.LOCKED_MESSAGE_TIME_TO_PEEK or 30 * 1000
+
+    start: ->
+      debug "Starting peek message every #{@timeToPeek} seconds... #{@_id()}"
+      @stream = @_stream().flatMap @_touch
+      @stream.resume()
+
+    delete: ->
+      debug "Deleting message #{@_id()}"
+      @service.deleteMessageAsync @lockedMessage
+
+    stop: ->
+      debug "Stopping peek message... #{@_id()}"
+      @stream.done => clearInterval @interval
+      @stream.end()
+
+    _touch: =>
+      debug "Touching message... #{@_id()}"
+      highland @service.renewLockForMessageAsync(@lockedMessage)
+
+    _stream: ->
+      highland (push, next) =>
+        @interval = setInterval (=> push undefined, 1), @timeToPeek
+
+    _id: => @lockedMessage.brokerProperties?.MessageId
```

Comment:
Por qué es calculada? puede variar @lockedMessage?

Reply:
Mmm, no entendí. No camba. Pero no quería asignarla al this, por si en algún momento quiero agregarle algo.